src/sinks/new_relic.rs

```diff
@@ -0,0 +1,890 @@
+                                    },
+                                    serde_json::Value::Number(n) => {
+                                        if n.is_f64() {
+                                            event_model.insert(k, Value::from(n.as_f64()));
+                                        }
+                                        else {
+                                            event_model.insert(k, Value::from(n.as_i64()));
+                                        }
+                                    },
+                                    serde_json::Value::Bool(b) => {
+                                        event_model.insert(k, Value::from(b));
+                                    },
+                                    _ => {}
+                                }
+                            }
+                            event_model.remove("message");
+                        }
+                    }
+
+                    if let None = event_model.get("eventType") {
+                        event_model.insert("eventType".to_owned(), Value::from("VectorSink".to_owned()));
+                    }
+
+                    events_array.push(event_model);
+                },
+                _ => {
+                    // Unrecognized event type
+                }
+            }
+        }
+
+        if events_array.len() > 0 {
+            Ok(Self::new(events_array))
+        }
+        else {
+            Err("No valid events to generate")
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct LogsApiModel(Vec<DataStore>);
+
+impl LogsApiModel {
+    pub fn new(logs_array: Vec<KeyValData>) -> Self {
+        let mut logs_store = DataStore::new();
+        logs_store.insert("logs".to_owned(), logs_array);
+        Self(vec!(logs_store))
+    }
+}
+
+impl ToJSON for LogsApiModel {}
+
+impl TryFrom<Vec<Event>> for LogsApiModel {
+    type Error = &'static str;
+
+    fn try_from(buf_events: Vec<Event>) -> Result<Self, Self::Error> {
+        let mut logs_array = vec!();
+        for buf_event in buf_events {
+            match buf_event {
+                Event::Log(log) => {
+                    let mut log_model = KeyValData::new();
+                    for (k, v) in log.all_fields() {
+                        log_model.insert(k, v.clone());
+                    }
+                    if let None = log.get("message") {
+                        log_model.insert("message".to_owned(), Value::from("log from vector".to_owned()));
+                    }
+                    logs_array.push(log_model);
+                },
+                _ => {
+                    // Unrecognized event type
+                }
+            }
+        }
+
+        if logs_array.len() > 0 {
+            Ok(Self::new(logs_array))
+        }
+        else {
+            Err("No valid logs to generate")
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct NewRelicSinkError {
```

Comment:
Using `vector_core::Error` instead of `NewRelicSinkError` inside `impl RetryLogic for NewRelicApiRetry` gives a compile error: "the trait \`Sized\` is not implemented for...".